core/arch/arm/mm/tee_pager.c

```diff
@@ -1158,8 +1158,15 @@ bool tee_pager_set_um_region_attr(struct user_mode_ctx *uctx, vaddr_t base,
 			if (flags & TEE_MATTR_UX) {
 				void *va = (void *)tblidx2va(tblidx);
 
+			/*
+			 * For BTI enabled binaries, the executable pages
+			 * may need to be remapped with Gaurded bit. So
+			 * we will hit this assertion.
+			 */
+#if !defined(CONFIG_BTI)
```

Comment:
This is not good enough. However, I believe this can be skipped if `a` already has TEE_MATTR_UX, provided that the BTI bit is cached in the TLB entry and not in the icache.